tests/Pagination/LengthAwarePaginatorTest.php

```diff
@@ -124,4 +124,11 @@ public function testItRetrievesThePaginatorOptions()
     {
         $this->assertSame($this->options, $this->p->getOptions());
     }
+
+    public function testLengthAwarePaginatorShouldAllowForZeroPerPage()
+    {
+        $p = new LengthAwarePaginator($array = ['item1', 'item2', 'item3', 'item4'], 4, 0, 2, $this->options);
+
+        $this->assertSame(0, $p->toArray()['per_Page']);
```

Comment:
Why not use the methods that are readily available?

```
$this->assertSame(0, $p->perPage());
// $this->assertSame(?, $p->lastPage());
// etc.
```

It would also help if you showed an example in your PR with how you are using this code.

Reply:
Locally we are using query builder and retrieving results directly back from our database. As far as operating directly against last page, my goal was to stop the application from dying on a 0.

I am currently learning my way around unit tests however, Is there a particular reason I should assert on lastPage over perPage?

Reply:
Made the suggested change, also realized that the member perPage not keeping value might be an unwelcome change for users. Keeps 0 value, avoids 0 division now.